gzip.go

```diff
@@ -28,12 +29,28 @@ func (g *gzipWriter) WriteString(s string) (int, error) {
 }
 
 func (g *gzipWriter) Write(data []byte) (int, error) {
+	g.gzipped = g.isGzipped(data) || g.gzipped
+	if g.gzipped {
+		return g.ResponseWriter.Write(data)
+	}
+
+	g.Header().Set("Content-Encoding", "gzip")
+	g.Header().Set("Vary", "Accept-Encoding")
 	g.Header().Del("Content-Length")
 	return g.writer.Write(data)
 }
 
 // Fix: https://github.com/mholt/caddy/issues/38
 func (g *gzipWriter) WriteHeader(code int) {
+	g.Header().Set("Content-Encoding", "gzip")
+	g.Header().Set("Vary", "Accept-Encoding")
 	g.Header().Del("Content-Length")
 	g.ResponseWriter.WriteHeader(code)
 }
+
+func (g *gzipWriter) isGzipped(input []byte) bool {
+	if len(input) < 2 {
+		return false
+	}
+	return input[0] == 0x1f && input[1] == 0x8b
```

Comment:
If you look at the `func (z *Reader) readHeader() (hdr Header, err error)`  on `compress/gzip/gunzip.go` file it should include the deflate option too, hence this should do it:
```go
var gzipHeader = []byte{0x1f, 0x8b, 0x08}

func IsGzip(b []byte) bool {
        if len(b) < 3 {
                return false
        }

        return bytes.Equal(b[0:3], gzipHeader)
}
```